### PR TITLE
chore: git mod - connect modal and branches

### LINKED ISSUE #37802: [Task] Git mod - Components - Connect modal

### LINKED ISSUE #37814: [Task] Git mod - Components - Error states

### PR DESCRIPTION
**Pull request by brayn003**
## Description
- Connects connect modal to store
- Adds component for branch list

Fixes #37814
Fixes #37802 

## Automation

/ok-to-test tags="@tag.Git"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/12376713895>
> Commit: 34d96b492e20118aa5d82bfa10b9f09315b2bc60
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=12376713895&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Git`
> Spec:
> <hr>Tue, 17 Dec 2024 16:59:28 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a `ConnectInitialize` component for Git provider connections.
	- Added `ConnectSuccess` modal to display successful Git connections.
	- Implemented `ImportModal` for Git import functionality.
	- Added `LocalProfileView` for managing user profile settings.
	- Introduced `BranchList` component for displaying and managing branches.

- **Bug Fixes**
	- Enhanced error handling and state management in various components.

- **Refactor**
	- Streamlined prop handling and state management in multiple components.
	- Updated action and selector names for clarity and consistency.

- **Tests**
	- Expanded test coverage for new and modified components, ensuring accurate behavior under various conditions.

- **Chores**
	- Updated import paths and removed unused files to improve project structure.

### CODE REVIEW

File: app/client/src/git/components/DefaultBranch/DefaultBranchCE.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+export default function DefaultBranchCE() {
+  const { branches } = useBranches();
+
+  return (
+    <DefaultBranchView
+      branches={branches}
+      isGitProtectedFeatureLicensed={false}
+    />
+  );
+}
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/git/components/DefaultBranch/DefaultBranchCE.tsx:14]: this is fine, the check is at an upper level

File: app/client/src/git/components/GitContextProvider/index.tsx [hunk 2]
```diff
@@ -13,17 +26,65 @@ export const useGitContext = () => {
...
+  const setImportWorkspaceId = useCallback(() => {
+    if (workspaceId) {
+      dispatch(
+        setWorkspaceIdForImport({
+          workspaceId: workspaceId ?? "",
+          editorId: artifactId ?? "",
+        }),
+      );
+    }
+  }, [artifactId, dispatch, setWorkspaceIdForImport, workspaceId]);
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/git/components/GitContextProvider/index.tsx:70]: don;t want to throw when dispatching actions

File: app/client/src/git/ce/components/GitModals/index.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import ConflictErrorModal from "git/components/ConflictErrorModal";
+import ConnectModal from "git/components/ConnectModal";
+import DisableAutocommitModal from "git/components/DisableAutocommitModal";
+import DisconnectModal from "git/components/DisconnectModal";
+import OpsModal from "git/components/OpsModal";
+import SettingsModal from "git/components/SettingsModal";
+import React from "react";
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/git/ce/components/GitModals/index.tsx:7]: But these are correct, you seem to be hallucinating

File: app/client/src/git/components/BranchList/index.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    isCreateBranchLoading,
+    isFetchBranchesLoading,
+    toggleBranchPopup,
+  } = useBranches();
+  const { defaultBranch } = useDefaultBranch();
+  const {
+    fetchProtectedBranches,
+    isFetchProtectedBranchesLoading,
+    protectedBranches,
+  } = useProtectedBranches();
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/git/components/BranchList/index.tsx:26]: Not true in every case

File: app/client/src/git/ce/store/types.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export interface GitArtifactAPIResponsesReduxState {}
+
+export interface GitArtifactUIReduxState {}
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/git/ce/store/types.ts:3]: why?

File: app/client/src/git/ce/components/DefaultBranch/DefaultBranchView.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+  const handleUpdate = useCallback(() => {
+    if (selectedValue) {
+      AnalyticsUtil.logEvent("GS_DEFAULT_BRANCH_UPDATE", {
+        old_branch: currentDefaultBranch,
+        new_branch: selectedValue,
+      });
+      updateDefaultBranch(selectedValue);
+    }
+  }, [currentDefaultBranch, selectedValue, updateDefaultBranch]);
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/git/ce/components/DefaultBranch/DefaultBranchView.tsx:100]: action creators don't throw error

File: app/client/src/git/ce/store/types.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/git/ce/store/types.ts:3]: I don't think an empty interface doesn't allow just any property. Do you have documentation link to prove this?

**Review by ashit-rath — APPROVED** [review]

**Pull request merged**